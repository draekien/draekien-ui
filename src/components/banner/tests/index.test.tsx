import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Banner from '..';

describe('Banner component', () => {
  test('should match snapshot', () => {
    const { container } = render(<Banner message="banner" />);

    expect(container).toMatchSnapshot();
  });

  test('should match snapshot when all props are populated', () => {
    const { container } = render(
      <Banner
        variant="error"
        message="banner"
        width="20rem"
        icon="accessibility"
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('should call onClose if callback function passed and close button clicked', () => {
    const fn = jest.fn();

    const { container } = render(
      <Banner message="banner" closable onClose={fn} />
    );

    fireEvent(
      container.querySelectorAll('i')[1],
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );

    expect(fn).toHaveBeenCalledTimes(1);
  });
});
