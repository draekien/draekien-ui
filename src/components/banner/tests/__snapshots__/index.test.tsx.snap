// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component should match snapshot 1`] = `
<div>
  <div
    class="css-1qxp57m-Banner"
  >
    <i
      class="material-icons css-1q3mgzl-Banner"
    >
      info
    </i>
    <div
      class="css-hu5ugq-Banner"
    >
      <div
        class="css-1s3pmdg-Text"
      >
        banner
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner component should match snapshot when all props are populated 1`] = `
<div>
  <div
    class="css-1uzbnv-Banner"
  >
    <i
      class="material-icons css-1vuensh-Banner"
    >
      accessibility
    </i>
    <div
      class="css-hu5ugq-Banner"
    >
      <div
        class="css-1s3pmdg-Text"
      >
        banner
      </div>
    </div>
  </div>
</div>
`;
